### PR TITLE
[ISSUE-162] Metrics proposal

### LINKED ISSUE #162: Prepare metrics proposal
1. Describe requirements (HTTP endpoints).
2. Prepare a proposal and make POC with the private monitoring system.

### PR DESCRIPTION
**Pull request by painhardcore**
## Purpose
Proposal for metrics of the Bare-metal CSI Plugin

## PR checklist
- [x] Add link to the issue
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved
- [ ] PR validation is passed
- [ ] Custom CI is passed

Closes #162

### CODE REVIEW

File: docs/proposals/metrics.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+## Proposal
+
+Expose metrics in Prometheus format with HTTP server at "/metrics" endpoint on every component:
+ - Node
+ - Controller  
+ - Scheduler Extender    
+ - Drive manager 
+
+The HTTP server will be optional since it can cause some security issues(i.e. vulnerabilities with HTTP server)
+Default port: 8888
```
Comment (rususlasan) [inline docs/proposals/metrics.md:21]: 8888 is acquired by drive manager

Comment (painhardcore) [reply docs/proposals/metrics.md:21]: Ok, changed to 8787.

**Review by rususlasan — APPROVED** [review]

File: docs/proposals/metrics.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ - Controller  
+ - Scheduler Extender    
+ - Drive manager 
+
+The HTTP server will be optional since it can cause some security issues(i.e. vulnerabilities with HTTP server)
+Default port: 8787
+
+Metrics:
+
+Metric name                           | Metric type |                          Labels/tags                                     |              Description
```
Comment (mishoyama) [inline docs/proposals/metrics.md:25]: @painhardcore is it possible to collect metrics from sidecar components - csi-provisioner, csi-attacher, etc?

Comment (painhardcore) [reply docs/proposals/metrics.md:25]: @mishoyama yes, as long they support metrics endpoint with Prometheus format.
From the first look - they have it.
[provisioned as example](https://github.com/kubernetes-csi/external-provisioner)
```
--metrics-address: The TCP network address where the prometheus metrics endpoint will run (example: :8080 which corresponds to port 8080 on local host). The default is empty string, which means metrics endpoint is disabled.

--metrics-path: The HTTP path where prometheus metrics will be exposed. Default is /metrics.
```

**Review by mishoyama — APPROVED** [review]

**Pull request merged**